cmd/restic/global.go

```diff
@@ -550,10 +550,18 @@ func parseConfig(loc location.Location, opts options.Options) (interface{}, erro
 			cfg.KeyID = os.Getenv("AWS_ACCESS_KEY_ID")
 		}
 
+		if cfg.KeyID == "" {
```

Comment:
This is not the right way to implement this check. The S3 backend supports multiple ways to pass credentials, see `internal/backend/s3/s3.go`.

What could work is checking that when a `KeyID` is specified there's also a `Secret` and vice versa.